checker/examples/Makefile

```diff
@@ -4,7 +4,6 @@ all: compile
 
 compile:
 	$(JAVAC) *.java
-	$(JAVAC) units-extension/*.java
```

Comment:
We should discuss what the purpose of the top-level examples/Makefile is.
At the moment it just compiles the files and doesn't check for expected errors - not a solid test suite.
In
https://github.com/typetools/checker-framework/pull/501/files
we're adding another subdirectory to the examples.
One option is that the top-level makefile compiles the current directory files and delegates to all sub-projects. The other option, which Jeff seems to be thinking of, is that the top-level only handles the files in the current directory and ignores sub-projects.